src/s390x/dasd.rs

```diff
@@ -0,0 +1,284 @@
+use crate::cmdline::*;
+use crate::download::write_image;
+use crate::errors::*;
+use crate::source::*;
+use dd_lib::dd;
+use dd_lib::opts::Opts;
+use fdisk::Context;
+use std::fs::{read_to_string, write, File};
+use std::path::{Path, PathBuf};
+use std::process::{Command, ExitStatus};
+use std::str::FromStr;
+
+/////////////////////////////////////////////////////////////////////////////
+// IBM DASD Support
+/////////////////////////////////////////////////////////////////////////////
+
+/// Format the disk and copy the image to it
+pub fn write_image_s390(config: &InstallConfig, source: &mut ImageSource) -> Result<()> {
+    eprintln!("Installing CoreOS to {}", config.device);
+    let image_filename = save_image(source)?;
+    low_level_format(&config.device)?;
+    if is_invalid(&config.device)? {
+        default_format(&config.device)?
+    }
+    let (options, partitions) = partitioning_options(&config.device, &image_filename)?;
+    make_partitions(&config.device, &partitions)?;
+    copy_partitions(&options)?;
+    Ok(())
+}
+
+/// Check process exit status
+pub fn check(status: ExitStatus) -> Result<()> {
+    if !status.success() {
+        return Err("Execution failed".into());
+    }
+    match status.code() {
+        Some(code) => {
+            if code != 0 {
+                return Err(format!("Failed with code: {}", code).into());
+            }
+        }
+        None => {
+            return Err(("Process terminated by signal").into());
+        }
+    }
+    Ok(())
+}
+
+/// Download and unpack CoreOS image to file.
+/// Return path to raw image or error
+///
+/// # Arguments
+/// * `source` - CoreOS image source
+fn save_image(source: &mut ImageSource) -> Result<PathBuf> {
+    let mut image_filename = PathBuf::new();
+    image_filename.push("/tmp/");
```

Comment:
This requires us to have enough RAM for the entire uncompressed image, assuming `/tmp` is a tmpfs.

Is it actually necessary for us to prefetch the file?  It seems as though streaming should work: we can fetch enough of the image to get the partition table, do our partition table processing, and then stream each partition to disk in turn.

Reply:
i'll check it.

Reply:
so, i've checked truncated images - it doesn't work (fdisk fails with `GPT PMBR size mismatch`). we have to fetch the entire image